cpp/benchmarks/ndsh/utils.hpp

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/detail/nvtx/ranges.hpp>
+#include <cudf/groupby.hpp>
+#include <cudf/io/parquet.hpp>
+
+#include <rmm/device_uvector.hpp>
+
+// RMM memory resource creation utilities
+auto make_cuda();
+auto make_async();
+auto make_pool();
+auto make_managed();
+auto make_managed_pool();
+auto make_prefetch();
+auto make_prefetch_pool();
+
+/**
+ * @brief Create an RMM memory resource
+ *
+ * @param rmm_mode The type of memory resource to create
+ */
+std::shared_ptr<rmm::mr::device_memory_resource> create_memory_resource(
+  std::string const& rmm_mode);

```

Comment:
All this memory resource logic is very similar to `cpp/benchmarks/fixture/nvbench_fixture.hpp`. Can we reuse that?

However, I don't see where `create_memory_resource` is being called now that these are NVBench benchmarks -- is this code still needed at all?